Microsoft.Build.NoTargets broken for MSBuild 15

Commit 38f7f902c291d41af120fd1640a2691bd329011d has "broken" `.msbuildproj` projects, as it does not correctly import the manged targets for `.msbuildproj` projects.

See:
https://github.com/microsoft/MSBuildSdks/blob/38f7f902c291d41af120fd1640a2691bd329011d/src/NoTargets/Sdk/Sdk.targets#L74-L78

This has resulted in errors like 
`C:\Users\NoOne\.nuget\packages\microsoft.build.notargets\1.0.88\Sdk\Sdk.targets(78,3): error MSB4019: The imported project "C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\MSBuild\15.0\Bin\Microsoft.Managed.targets" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk. [C:\B\A3\_work\16\s\src\SomeProject.msbuildproj]`

### Possible fix

As the `Microsoft.Managed.targets` file does not exist in earlier tool versions, and you only need it to support GraphBuild, one solution would be to import `Microsoft.Managed.targets` conditionally.

```xml
 <!-- 
    Microsoft.Managed.Targets is imported by the managed language target files, but most of the msbuild tasks are actually in Microsoft.Common.Currentversion.targets.
    So import it when the managed targets do not get imported, to support GraphBuild
  -->
  <Import Project="$(MSBuildToolsPath)\Microsoft.Managed.targets" Condition="'$(ManagedLanguageTargetsGotImported)' != 'true' and Exists('$(MSBuildToolsPath)\Microsoft.Managed.targets')"/>
```

@jeffkl  I have run your little repro project, and I see the same issue:

### Run
```
**********************************************************************
** Visual Studio 2017 Developer Command Prompt v15.9.13
** Copyright (c) 2017 Microsoft Corporation
**********************************************************************

C:\NoTargetRepro>msbuild notargets.msbuildproj
Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 1/23/2020 7:36:15 AM.
Project "C:\NoTargetRepro\notargets.msbuildproj" on node 1 (default targets).
C:\Users\JOHNDOE\.nuget\packages\microsoft.build.notargets\1.0.88\Sdk\Sdk.targets(78,3): error MSB4019: The imported project "C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\MSBuild\15.0\bin\Microsoft.Managed.targets" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk. [C:\NoTargetRepro\notargets.msbuildproj]
Done Building Project "C:\NoTargetRepro\notargets.msbuildproj" (default targets) -- FAILED.


Build FAILED.

"C:\NoTargetRepro\notargets.msbuildproj" (default target) (1) ->
  C:\Users\JOHNDOE\.nuget\packages\microsoft.build.notargets\1.0.88\Sdk\Sdk.targets(78,3): error MSB4019: The imported project "C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\MSBuild\15.0\bin\Microsoft.Managed.targets" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk. [C:\NoTargetRepro\notargets.msbuildproj]

    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:00.48

C:\NoTargetRepro>
```

### Imports
```
C
    NoTargetRepro
        notargets.msbuildproj
    Program Files (x86)\Microsoft Visual Studio\2017\Professional
        Common7\IDE\CommonExtensions\Microsoft\NuGet
            NuGet.targets
        MSBuild
            15.0
                bin
                    Microsoft.Common.CurrentVersion.targets
                    Microsoft.Common.targets
                    Microsoft.Data.Entity.targets
                    Microsoft.NETFramework.CurrentVersion.props
                    Microsoft.NETFramework.CurrentVersion.targets
                    Microsoft.NETFramework.props
                    Microsoft.NETFramework.targets
                    Microsoft.WinFX.targets
                    Microsoft.WorkflowBuildExtensions.targets
                    Microsoft.Xaml.targets
                Imports\Microsoft.Common.props\ImportBefore
                    Microsoft.NuGet.ImportBefore.props
                Microsoft.Common.props
                Microsoft.Common.targets
                    ImportAfter
                        Microsoft.DesktopBridge.ImportAfter.targets
                        Microsoft.Docker.ImportAfter.targets
                        Microsoft.NET.Build.Extensions.targets
                        Microsoft.Net.CoreRuntime.ImportAfter.targets
                        Microsoft.NetNative.ImportAfter.targets
                        Microsoft.NuGet.ImportAfter.targets
                        Microsoft.Web.ImportAfter.targets
                    ImportBefore
                        Microsoft.Cpp.VCLibs120Universal.targets
                        Microsoft.Net.CoreRuntime.ImportBefore.targets
                        Microsoft.NetNative.ImportBefore.targets
            Microsoft
                DesktopBridge
                    Microsoft.DesktopBridge.Common.targets
                Microsoft.NET.Build.Extensions
                    Microsoft.NET.Build.Extensions.NETFramework.targets
                    Microsoft.NET.Build.Extensions.targets
                NuGet\15.0
                    Microsoft.NuGet.props
                VisualStudio
                    Managed
                        Microsoft.Managed.DesignTime.targets
                    v15.0
                        CodeAnalysis
                            Microsoft.CodeAnalysis.targets
                        TeamTest
                            Microsoft.TeamTest.targets
            Sdks\Microsoft.Docker.Sdk\build
                Microsoft.Docker.targets
    Program Files\dotnet\sdk\2.1.507
        Microsoft.NETCoreSdk.BundledCliTools.props
        Microsoft.NETCoreSdk.BundledVersions.props
        Sdks
            Microsoft.NET.Sdk
                Sdk
                    Sdk.props
                    Sdk.targets
                targets
                    Microsoft.NET.ComposeStore.targets
                    Microsoft.NET.ConflictResolution.targets
                    Microsoft.NET.CrossGen.targets
                    Microsoft.NET.DefaultAssemblyInfo.targets
                    Microsoft.NET.DefaultOutputPaths.targets
                    Microsoft.NET.DefaultPackageConflictOverrides.targets
                    Microsoft.NET.GenerateAssemblyInfo.targets
                    Microsoft.NET.NuGetOfflineCache.targets
                    Microsoft.NET.ObsoleteReferences.targets
                    Microsoft.NET.PackTool.props
                    Microsoft.NET.PackTool.targets
                    Microsoft.NET.PreserveCompilationContext.targets
                    Microsoft.NET.Publish.targets
                    Microsoft.NET.RuntimeIdentifierInference.targets
                    Microsoft.NET.Sdk.BeforeCommon.targets
                    Microsoft.NET.Sdk.Common.targets
                    Microsoft.NET.Sdk.DefaultItems.props
                    Microsoft.NET.Sdk.DefaultItems.targets
                    Microsoft.NET.Sdk.props
                    Microsoft.NET.Sdk.targets
                    Microsoft.NET.SupportedTargetFrameworks.props
                    Microsoft.NET.TargetFrameworkInference.targets
                    Microsoft.PackageDependencyResolution.targets
            NuGet.Build.Tasks.Pack\build
                NuGet.Build.Tasks.Pack.targets
    Users\JOHNDOE\.nuget\packages\microsoft.build.notargets\1.0.88\Sdk
        Sdk.props
        Sdk.targets
    Windows\Microsoft.NET\Framework\v4.0.30319
        Microsoft.Data.Entity.targets
        Microsoft.WinFx.targets
        Microsoft.WorkflowBuildExtensions.targets
        Microsoft.Xaml.targets
```
#### Imported Targets
```
C
    NoTargetRepro
        notargets.msbuildproj
    Program Files (x86)\Microsoft Visual Studio\2017\Professional
        Common7\IDE\CommonExtensions\Microsoft\NuGet
            NuGet.targets
                Target Name=Restore Project=
                Target Name=GenerateRestoreGraphFile Project=
                Target Name=CollectPackageReferences Project=
                Target Name=_LoadRestoreGraphEntryPoints Project=
                Target Name=_FilterRestoreGraphProjectInputItems Project=
                Target Name=_GenerateRestoreGraph Project=
                Target Name=_GenerateRestoreGraphProjectEntry Project=
                Target Name=_GenerateRestoreSpecs Project=
                Target Name=_GenerateDotnetCliToolReferenceSpecs Project=
                Target Name=_GetProjectJsonPath Project=
                Target Name=_GetRestoreProjectStyle Project=
                Target Name=EnableIntermediateOutputPathMismatchWarning Project=
                Target Name=_GetRestoreTargetFrameworksOutput Project=
                Target Name=_GetRestoreTargetFrameworksAsItems Project=
                Target Name=_GetRestoreSettings Project=
                Target Name=_GetRestoreSettingsCurrentProject Project=
                Target Name=_GetRestoreSettingsAllFrameworks Project=
                Target Name=_GetRestoreSettingsPerFramework Project=
                Target Name=_GenerateRestoreProjectSpec Project=
                Target Name=_GenerateProjectRestoreGraph Project=
                Target Name=_GenerateRestoreDependencies Project=
                Target Name=_GenerateProjectRestoreGraphAllFrameworks Project=
                Target Name=_GenerateProjectRestoreGraphCurrentProject Project=
                Target Name=_GenerateProjectRestoreGraphPerFramework Project=
                Target Name=_GenerateRestoreProjectPathItemsCurrentProject Project=
                Target Name=_GenerateRestoreProjectPathItemsPerFramework Project=
                Target Name=_GenerateRestoreProjectPathItems Project=
                Target Name=_GenerateRestoreProjectPathItemsAllFrameworks Project=
                Target Name=_GenerateRestoreProjectPathWalk Project=
                Target Name=_GetAllRestoreProjectPathItems Project=
                Target Name=_GetRestoreSettingsOverrides Project=
                Target Name=_GetRestorePackagesPathOverride Project=
                Target Name=_GetRestoreSourcesOverride Project=
                Target Name=_GetRestoreFallbackFoldersOverride Project=
                Target Name=_IsProjectRestoreSupported Project=
        MSBuild
            15.0
                bin
                    Microsoft.Common.CurrentVersion.targets
                        Target Name=InitializeSourceControlInformation Project=
                        Target Name=_CheckForInvalidConfigurationAndPlatform Project=
                        Target Name=Build Project=
                        Target Name=BeforeBuild Project=
                        Target Name=AfterBuild Project=
                        Target Name=CoreBuild Project=
                        Target Name=Rebuild Project=
                        Target Name=BeforeRebuild Project=
                        Target Name=AfterRebuild Project=
                        Target Name=BuildGenerateSources Project=
                        Target Name=BuildGenerateSourcesTraverse Project=
                        Target Name=BuildCompile Project=
                        Target Name=BuildCompileTraverse Project=
                        Target Name=BuildLink Project=
                        Target Name=BuildLinkTraverse Project=
                        Target Name=CopyRunEnvironmentFiles Project=
                        Target Name=Run Project=
                        Target Name=BuildOnlySettings Project=
                        Target Name=PrepareForBuild Project=
                        Target Name=GetFrameworkPaths Project=
                        Target Name=GetReferenceAssemblyPaths Project=
                        Target Name=GetTargetFrameworkMoniker Project=
                        Target Name=GetTargetFrameworkMonikerDisplayName Project=
                        Target Name=GetTargetFrameworkDirectories Project=
                        Target Name=AssignLinkMetadata Project=
                        Target Name=PreBuildEvent Project=
                        Target Name=UnmanagedUnregistration Project=
                        Target Name=GetTargetFrameworkVersion Project=
                        Target Name=ResolveReferences Project=
                        Target Name=BeforeResolveReferences Project=
                        Target Name=AfterResolveReferences Project=
                        Target Name=AssignProjectConfiguration Project=
                        Target Name=_SplitProjectReferencesByFileExistence Project=
                        Target Name=_GetProjectReferenceTargetFrameworkProperties Project=
                        Target Name=GetTargetFrameworks Project=
                        Target Name=GetTargetFrameworkProperties Project=
                        Target Name=PrepareProjectReferences Project=
                        Target Name=ResolveProjectReferences Project=
                        Target Name=ResolveProjectReferencesDesignTime Project=
                        Target Name=ExpandSDKReferencesDesignTime Project=
                        Target Name=GetTargetPath Project=
                        Target Name=GetTargetPathWithTargetPlatformMoniker Project=
                        Target Name=GetNativeManifest Project=
                        Target Name=ResolveNativeReferences Project=
                        Target Name=ResolveAssemblyReferences Project=
                        Target Name=FindReferenceAssembliesForReferences Project=
                        Target Name=GenerateBindingRedirects Project=
                        Target Name=GenerateBindingRedirectsUpdateAppConfig Project=
                        Target Name=GetInstalledSDKLocations Project=
                        Target Name=ResolveSDKReferences Project=
                        Target Name=ResolveSDKReferencesDesignTime Project=
                        Target Name=FindInvalidProjectReferences Project=
                        Target Name=GetReferenceTargetPlatformMonikers Project=
                        Target Name=ExpandSDKReferences Project=
                        Target Name=ExportWindowsMDFile Project=
                        Target Name=ResolveAssemblyReferencesDesignTime Project=
                        Target Name=DesignTimeResolveAssemblyReferences Project=
                        Target Name=ResolveComReferences Project=
                        Target Name=ResolveComReferencesDesignTime Project=
                        Target Name=PrepareResources Project=
                        Target Name=PrepareResourceNames Project=
                        Target Name=AssignTargetPaths Project=
                        Target Name=GetItemTargetPaths Project=
                        Target Name=SplitResourcesByCulture Project=
                        Target Name=CreateCustomManifestResourceNames Project=
                        Target Name=ResGen Project=
                        Target Name=BeforeResGen Project=
                        Target Name=AfterResGen Project=
                        Target Name=CoreResGen Project=
                        Target Name=CompileLicxFiles Project=
                        Target Name=ResolveKeySource Project=
                        Target Name=Compile Project=
                        Target Name=_GenerateCompileInputs Project=
                        Target Name=GenerateTargetFrameworkMonikerAttribute Project=
                        Target Name=GenerateAdditionalSources Project=
                        Target Name=BeforeCompile Project=
                        Target Name=AfterCompile Project=
                        Target Name=_TimeStampBeforeCompile Project=
                        Target Name=_GenerateCompileDependencyCache Project=
                        Target Name=_TimeStampAfterCompile Project=
                        Target Name=_ComputeNonExistentFileProperty Project=
                        Target Name=GenerateSerializationAssemblies Project=
                        Target Name=CreateSatelliteAssemblies Project=
                        Target Name=_GenerateSatelliteAssemblyInputs Project=
                        Target Name=GenerateSatelliteAssemblies Project=
                        Target Name=ComputeIntermediateSatelliteAssemblies Project=
                        Target Name=SetWin32ManifestProperties Project=
                        Target Name=_SetExternalWin32ManifestProperties Project=
                        Target Name=_SetEmbeddedWin32ManifestProperties Project=
                        Target Name=_GenerateResolvedDeploymentManifestEntryPoint Project=
                        Target Name=GenerateManifests Project=
                        Target Name=GenerateApplicationManifest Project=
                        Target Name=_DeploymentComputeNativeManifestInfo Project=
                        Target Name=_DeploymentComputeClickOnceManifestInfo Project=
                        Target Name=_DeploymentGenerateTrustInfo Project=
                        Target Name=GenerateDeploymentManifest Project=
                        Target Name=PrepareForRun Project=
                        Target Name=CopyFilesToOutputDirectory Project=
                        Target Name=_CopyFilesMarkedCopyLocal Project=
                        Target Name=_CopySourceItemsToOutputDirectory Project=
                        Target Name=GetCopyToOutputDirectoryItems Project=
                        Target Name=GetCopyToPublishDirectoryItems Project=
                        Target Name=_CopyOutOfDateSourceItemsToOutputDirectory Project=
                        Target Name=_CopyOutOfDateSourceItemsToOutputDirectoryAlways Project=
                        Target Name=_CopyAppConfigFile Project=
                        Target Name=_CopyManifestFiles Project=
                        Target Name=_CheckForCompileOutputs Project=
                        Target Name=_SGenCheckForOutputs Project=
                        Target Name=UnmanagedRegistration Project=
                        Target Name=IncrementalClean Project=
                        Target Name=_CleanGetCurrentAndPriorFileWrites Project=
                        Target Name=Clean Project=
                        Target Name=BeforeClean Project=
                        Target Name=AfterClean Project=
                        Target Name=CleanReferencedProjects Project=
                        Target Name=CoreClean Project=
                        Target Name=_CleanRecordFileWrites Project=
                        Target Name=CleanPublishFolder Project=
                        Target Name=PostBuildEvent Project=
                        Target Name=Publish Project=
                        Target Name=_DeploymentUnpublishable Project=
                        Target Name=SetGenerateManifests Project=
                        Target Name=PublishOnly Project=
                        Target Name=BeforePublish Project=
                        Target Name=AfterPublish Project=
                        Target Name=PublishBuild Project=
                        Target Name=_CopyFilesToPublishFolder Project=
                        Target Name=_DeploymentGenerateBootstrapper Project=
                        Target Name=_DeploymentSignClickOnceDeployment Project=
                        Target Name=AllProjectOutputGroups Project=
                        Target Name=BuiltProjectOutputGroup Project=
                        Target Name=DebugSymbolsProjectOutputGroup Project=
                        Target Name=DocumentationProjectOutputGroup Project=
                        Target Name=SatelliteDllsProjectOutputGroup Project=
                        Target Name=SourceFilesProjectOutputGroup Project=
                        Target Name=GetCompile Project=
                        Target Name=ContentFilesProjectOutputGroup Project=
                        Target Name=SGenFilesOutputGroup Project=
                        Target Name=GetResolvedSDKReferences Project=
                        Target Name=CollectReferencedNuGetPackages Project=
                        Target Name=PriFilesOutputGroup Project=
                        Target Name=SDKRedistOutputGroup Project=
                        Target Name=AllProjectOutputGroupsDependencies Project=
                        Target Name=BuiltProjectOutputGroupDependencies Project=
                        Target Name=DebugSymbolsProjectOutputGroupDependencies Project=
                        Target Name=SatelliteDllsProjectOutputGroupDependencies Project=
                        Target Name=DocumentationProjectOutputGroupDependencies Project=
                        Target Name=SGenFilesOutputGroupDependencies Project=
                        Target Name=ReferenceCopyLocalPathsOutputGroup Project=
                        Target Name=CleanAppxPackage Project=
                        Target Name=GetPackagingOutputs Project=
                    Microsoft.Common.targets
                    Microsoft.Data.Entity.targets
                    Microsoft.NETFramework.CurrentVersion.props
                    Microsoft.NETFramework.CurrentVersion.targets
                        Target Name=GetFrameworkPaths Project=
                        Target Name=ImplicitlyExpandDesignTimeFacades Project=
                    Microsoft.NETFramework.props
                    Microsoft.NETFramework.targets
                    Microsoft.WinFX.targets
                    Microsoft.WorkflowBuildExtensions.targets
                    Microsoft.Xaml.targets
                Imports\Microsoft.Common.props\ImportBefore
                    Microsoft.NuGet.ImportBefore.props
                Microsoft.Common.props
                Microsoft.Common.targets
                    ImportAfter
                        Microsoft.DesktopBridge.ImportAfter.targets
                        Microsoft.Docker.ImportAfter.targets
                        Microsoft.NET.Build.Extensions.targets
                        Microsoft.Net.CoreRuntime.ImportAfter.targets
                        Microsoft.NetNative.ImportAfter.targets
                            Target Name=SetARM64AppxPackageInputsForInboxNetNative Project=
                        Microsoft.NuGet.ImportAfter.targets
                        Microsoft.Web.ImportAfter.targets
                    ImportBefore
                        Microsoft.Cpp.VCLibs120Universal.targets
                            Target Name=_FixVCLibs120References Project=
                            Target Name=_AddVCLibs140UniversalCrtDebugReference Project=
                        Microsoft.Net.CoreRuntime.ImportBefore.targets
                        Microsoft.NetNative.ImportBefore.targets
            Microsoft
                DesktopBridge
                    Microsoft.DesktopBridge.Common.targets
                        Target Name=DesktopBridgeCopyLocalOutputGroup Project=
                        Target Name=DesktopBridgeComFilesOutputGroup Project=
                        Target Name=GetDeployableContentReferenceOutputs Project=
                Microsoft.NET.Build.Extensions
                    Microsoft.NET.Build.Extensions.NETFramework.targets
                        Target Name=ImplicitlyExpandNETStandardFacades Project=
                        Target Name=_RemoveZipFileSuggestedRedirect Project=
                    Microsoft.NET.Build.Extensions.targets
                NuGet\15.0
                    Microsoft.NuGet.props
                VisualStudio
                    Managed
                        Microsoft.Managed.DesignTime.targets
                            Target Name=ResolvePackageDependenciesDesignTime Project=
                            Target Name=CollectSDKReferencesDesignTime Project=
                            Target Name=CollectResolvedSDKReferencesDesignTime Project=
                            Target Name=CollectPackageReferences Project=
                            Target Name=_CheckCompileDesignTimePrerequisite Project=
                            Target Name=CollectAnalyzersDesignTime Project=
                            Target Name=CollectResolvedCompilationReferencesDesignTime Project=
                            Target Name=CollectUpToDateCheckInputDesignTime Project=
                            Target Name=CollectUpToDateCheckOutputDesignTime Project=
                            Target Name=CollectUpToDateCheckBuiltDesignTime Project=
                    v15.0
                        CodeAnalysis
                            Microsoft.CodeAnalysis.targets
                                Target Name=SetCABuildNativeEnvironmentVariables Project=
                                Target Name=RunCodeAnalysis Project=
                                Target Name=RunNativeCodeAnalysis Project=
                                Target Name=RunSelectedFileNativeCodeAnalysis Project=
                                Target Name=RunMergeNativeCodeAnalysis Project=
                        TeamTest
                            Microsoft.TeamTest.targets
                                Target Name=ResolveTestReferences Project=
            Sdks\Microsoft.Docker.Sdk\build
                Microsoft.Docker.targets
                    Target Name=DockerResolveAppType Project=
                    Target Name=DockerUpdateComposeVsGeneratedFiles Project=
                    Target Name=DockerResolveTargetFramework Project=
                    Target Name=DockerComposeBuild Project=
                    Target Name=DockerPackageService Project=
    Program Files\dotnet\sdk\2.1.507
        Microsoft.NETCoreSdk.BundledCliTools.props
        Microsoft.NETCoreSdk.BundledVersions.props
        Sdks
            Microsoft.NET.Sdk
                Sdk
                    Sdk.props
                    Sdk.targets
                targets
                    Microsoft.NET.ComposeStore.targets
                        Target Name=ComposeStore Project=
                        Target Name=StoreWorkerMain Project=
                        Target Name=StoreWorkerMapper Project=
                        Target Name=StoreResolver Project=
                        Target Name=StoreWorkerPerformWork Project=
                        Target Name=StoreFinalizer Project=
                        Target Name=_CopyResolvedOptimizedFiles Project=
                        Target Name=PrepareForComposeStore Project=
                        Target Name=PrepforRestoreForComposeStore Project=
                        Target Name=RestoreForComposeStore Project=
                        Target Name=ComputeAndCopyFilesToStoreDirectory Project=
                        Target Name=CopyFilesToStoreDirectory Project=
                        Target Name=_CopyResolvedUnOptimizedFiles Project=
                        Target Name=_ComputeResolvedFilesToStoreTypes Project=
                        Target Name=_SplitResolvedFiles Project=
                        Target Name=_GetResolvedFilesToStore Project=
                        Target Name=ComputeFilesToStore Project=
                        Target Name=PrepRestoreForStoreProjects Project=
                    Microsoft.NET.ConflictResolution.targets
                        Target Name=_HandlePackageFileConflicts Project=
                        Target Name=_HandlePublishFileConflicts Project=
                    Microsoft.NET.CrossGen.targets
                        Target Name=PrepOptimizer Project=
                        Target Name=_RunOptimizer Project=
                        Target Name=RunCrossGen Project=
                        Target Name=_InitializeBasicProps Project=
                        Target Name=_GetCrossgenProps Project=
                        Target Name=_SetupStageForCrossgen Project=
                        Target Name=_RestoreCrossgen Project=
                    Microsoft.NET.DefaultAssemblyInfo.targets
                    Microsoft.NET.DefaultOutputPaths.targets
                    Microsoft.NET.DefaultPackageConflictOverrides.targets
                    Microsoft.NET.GenerateAssemblyInfo.targets
                        Target Name=GenerateAssemblyInfo Project=
                        Target Name=AddSourceRevisionToInformationalVersion Project=
                        Target Name=GetAssemblyAttributes Project=
                        Target Name=CreateGeneratedAssemblyInfoInputsCacheFile Project=
                        Target Name=CoreGenerateAssemblyInfo Project=
                        Target Name=GetAssemblyVersion Project=
                    Microsoft.NET.NuGetOfflineCache.targets
                    Microsoft.NET.ObsoleteReferences.targets
                        Target Name=_CheckForObsoleteDotNetCliToolReferences Project=
                    Microsoft.NET.PackTool.props
                    Microsoft.NET.PackTool.targets
                        Target Name=PackTool Project=
                        Target Name=GenerateToolsSettingsFileFromBuildProperty Project=
                        Target Name=ResolveApphostAsset Project=
                    Microsoft.NET.PreserveCompilationContext.targets
                        Target Name=ComputeDependencyFileCompilerOptions Project=
                        Target Name=ComputeRefAssembliesToPublish Project=
                        Target Name=_CopyReferenceOnlyAssembliesForBuild Project=
                    Microsoft.NET.Publish.targets
                        Target Name=_PublishBuildAlternative Project=
                        Target Name=_PublishNoBuildAlternative Project=
                        Target Name=Publish Project=
                        Target Name=_PreventProjectReferencesFromBuilding Project=
                        Target Name=PrepareForPublish Project=
                        Target Name=ComputeAndCopyFilesToPublishDirectory Project=
                        Target Name=CopyFilesToPublishDirectory Project=
                        Target Name=_CopyResolvedFilesToPublishPreserveNewest Project=
                        Target Name=_CopyResolvedFilesToPublishAlways Project=
                        Target Name=_ComputeResolvedFilesToPublishTypes Project=
                        Target Name=ComputeFilesToPublish Project=
                        Target Name=_ComputeNetPublishAssets Project=
                        Target Name=RunResolvePublishAssemblies Project=
                        Target Name=FilterPublishSatelliteResources Project=
                        Target Name=_ComputeCopyToPublishDirectoryItems Project=
                        Target Name=GetCopyToPublishDirectoryItems Project=
                        Target Name=DefaultCopyToPublishDirectoryMetadata Project=
                        Target Name=GeneratePublishDependencyFile Project=
                        Target Name=_ComputeExcludeFromPublishPackageReferences Project=
                        Target Name=_ParseTargetManifestFiles Project=
                        Target Name=GeneratePublishRuntimeConfigurationFile Project=
                        Target Name=DeployAppHost Project=
                    Microsoft.NET.RuntimeIdentifierInference.targets
                        Target Name=_CheckForUnsupportedAppHostUsage Project=
                        Target Name=_CheckForMismatchingPlatform Project=
                        Target Name=_CheckForNETCoreSdkIsPreview Project=
                        Target Name=AdjustDefaultPlatformTargetForNetFrameworkExeWithNoNativeCopyLocalItems Project=
                    Microsoft.NET.Sdk.BeforeCommon.targets
                    Microsoft.NET.Sdk.Common.targets
                        Target Name=GetTargetFrameworkProperties Project=
                    Microsoft.NET.Sdk.DefaultItems.props
                    Microsoft.NET.Sdk.DefaultItems.targets
                        Target Name=CheckForImplicitPackageReferenceOverrides Project=
                        Target Name=CheckForDuplicateItems Project=
                    Microsoft.NET.Sdk.props
                    Microsoft.NET.Sdk.targets
                        Target Name=GenerateBuildDependencyFile Project=
                        Target Name=GenerateBuildRuntimeConfigurationFiles Project=
                        Target Name=AddRuntimeConfigFileToBuiltProjectOutputGroupOutput Project=
                        Target Name=_SdkBeforeClean Project=
                        Target Name=_SdkBeforeRebuild Project=
                        Target Name=_ComputeNETCoreBuildOutputFiles Project=
                        Target Name=_ComputeReferenceAssemblies Project=
                        Target Name=CoreGenerateSatelliteAssemblies Project=
                        Target Name=_GetAssemblyInfoFromTemplateFile Project=
                        Target Name=_DefaultMicrosoftNETPlatformLibrary Project=
                        Target Name=GetAllRuntimeIdentifiers Project=
                    Microsoft.NET.SupportedTargetFrameworks.props
                    Microsoft.NET.TargetFrameworkInference.targets
                        Target Name=_CheckForUnsupportedTargetFramework Project=
                        Target Name=_CollectTargetFrameworkForTelemetry Project=
                        Target Name=_CheckForUnsupportedNETCoreVersion Project=
                        Target Name=_CheckForUnsupportedNETStandardVersion Project=
                    Microsoft.PackageDependencyResolution.targets
                        Target Name=ResolvePackageDependenciesForBuild Project=
                        Target Name=RunResolvePackageDependencies Project=
                        Target Name=ResolvePackageAssets Project=
                        Target Name=FilterSatelliteResources Project=
                        Target Name=ResolvePackageDependenciesDesignTime Project=
                        Target Name=CollectSDKReferencesDesignTime Project=
                        Target Name=CollectResolvedSDKReferencesDesignTime Project=
                        Target Name=RunProduceContentAssets Project=
                        Target Name=ReportAssetsLogMessages Project=
                        Target Name=ResolveLockFileReferences Project=
                        Target Name=IncludeTransitiveProjectReferences Project=
                        Target Name=ResolveLockFileAnalyzers Project=
                        Target Name=_ComputeLockFileCopyLocal Project=
                        Target Name=ResolveLockFileCopyLocalProjectDeps Project=
            NuGet.Build.Tasks.Pack\build
                NuGet.Build.Tasks.Pack.targets
                    Target Name=_GetOutputItemsFromPack Project=
                    Target Name=_GetTargetFrameworksOutput Project=
                    Target Name=_PackAsBuildAfterTarget Project=
                    Target Name=_CleanPackageFiles Project=
                    Target Name=_CalculateInputsOutputsForPack Project=
                    Target Name=Pack Project=
                    Target Name=_IntermediatePack Project=
                    Target Name=GenerateNuspec Project=
                    Target Name=_InitializeNuspecRepositoryInformationProperties Project=
                    Target Name=_LoadPackInputItems Project=
                    Target Name=_GetProjectReferenceVersions Project=
                    Target Name=_GetProjectVersion Project=
                    Target Name=_WalkEachTargetPerFramework Project=
                    Target Name=_GetFrameworksWithSuppressedDependencies Project=
                    Target Name=_GetFrameworkAssemblyReferences Project=
                    Target Name=_GetBuildOutputFilesWithTfm Project=
                    Target Name=_GetTfmSpecificContentForPackage Project=
                    Target Name=_GetDebugSymbolsWithTfm Project=
                    Target Name=_AddPriFileToPackBuildOutput Project=
                    Target Name=_GetPackageFiles Project=
    Users\JOHNDOE\.nuget\packages\microsoft.build.notargets\1.0.88\Sdk
        Sdk.props
        Sdk.targets
            Target Name=ComputeIntermediateSatelliteAssemblies Project=
            Target Name=GetTargetPath Project=
            Target Name=GetTargetPathWithTargetPlatformMoniker Project=
    Windows\Microsoft.NET\Framework\v4.0.30319
        Microsoft.Data.Entity.targets
            Target Name=EntityDeploy Project=
            Target Name=EntityDeploySplit Project=
            Target Name=EntityDeployNonEmbeddedResources Project=
            Target Name=EntityDeployEmbeddedResources Project=
            Target Name=EntityClean Project=
            Target Name=EntityDeploySetLogicalNames Project=
        Microsoft.WinFx.targets
            Target Name=GetWinFXPath Project=
            Target Name=DesignTimeMarkupCompilation Project=
            Target Name=PrepareResourcesForSatelliteAssemblies Project=
            Target Name=_AfterCompileWinFXInternal Project=
            Target Name=AfterCompileWinFX Project=
            Target Name=AfterMarkupCompilePass1 Project=
            Target Name=AfterMarkupCompilePass2 Project=
            Target Name=MarkupCompilePass1 Project=
            Target Name=MarkupCompilePass2 Project=
            Target Name=_CompileTemporaryAssembly Project=
            Target Name=MarkupCompilePass2ForMainAssembly Project=
            Target Name=GenerateTemporaryTargetAssembly Project=
            Target Name=CleanupTemporaryTargetAssembly Project=
            Target Name=AddIntermediateAssemblyToReferenceList Project=
            Target Name=SatelliteOnlyMarkupCompilePass2 Project=
            Target Name=HostInBrowserValidation Project=
            Target Name=SplashScreenValidation Project=
            Target Name=ResignApplicationManifest Project=
            Target Name=SignDeploymentManifest Project=
            Target Name=FileClassification Project=
            Target Name=MainResourcesGeneration Project=
            Target Name=SatelliteResourceGeneration Project=
            Target Name=GenerateResourceWithCultureItem Project=
            Target Name=SatelliteDllsProjectOutputGroup Project=
            Target Name=CheckUid Project=
            Target Name=UpdateUid Project=
            Target Name=RemoveUid Project=
            Target Name=MergeLocalizationDirectives Project=
            Target Name=AssignWinFXEmbeddedResource Project=
        Microsoft.WorkflowBuildExtensions.targets
            Target Name=ExpressionBuildExtension Project=
            Target Name=ValidationExtension Project=
            Target Name=GenerateCompiledExpressionsTempFile Project=
            Target Name=AddDeferredValidationErrorsFileToFileWrites Project=
            Target Name=ReportValidationBuildExtensionErrors Project=
            Target Name=DeferredValidation Project=
        Microsoft.Xaml.targets
            Target Name=DesignTimeXamlMarkupCompilation Project=
            Target Name=InProcessXamlMarkupCompilePass1 Project=
            Target Name=CleanInProcessXamlGeneratedFiles Project=
            Target Name=XamlMarkupCompileReadGeneratedFileList Project=
            Target Name=XamlMarkupCompilePass1 Project=
            Target Name=XamlMarkupCompileAddFilesGenerated Project=
            Target Name=XamlMarkupCompileReadPass2Flag Project=
            Target Name=XamlTemporaryAssemblyGeneration Project=
            Target Name=CompileTemporaryAssembly Project=
            Target Name=XamlMarkupCompilePass2 Project=
            Target Name=XamlMarkupCompileAddExtensionFilesGenerated Project=
            Target Name=GetCopyToOutputDirectoryXamlAppDefs Project=
```

### Environment
#### vswhere
`vswhere -prerelease -legacy`
```json
[
  {
    "instanceId": "51318366",
    "installDate": "2017-12-13T06:04:04Z",
    "installationName": "VisualStudio/15.9.13+28307.718",
    "installationPath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Professional",
    "installationVersion": "15.9.28307.718",
    "productId": "Microsoft.VisualStudio.Product.Professional",
    "productPath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Professional\\Common7\\IDE\\devenv.exe",
    "isPrerelease": false,
    "displayName": "Visual Studio Professional 2017",
    "description": "Professional developer tools and services for small teams",
    "channelId": "VisualStudio.15.Release",
    "channelPath": "C:\\Users\\JOHNDOE\\AppData\\Local\\Microsoft\\VisualStudio\\Packages\\_Channels\\4CB340F5\\catalog.json",
    "channelUri": "https://aka.ms/vs/15/release/channel",
    "enginePath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\Installer\\resources\\app\\ServiceHub\\Services\\Microsoft.VisualStudio.Setup.Service",
    "releaseNotes": "https://go.microsoft.com/fwlink/?LinkId=660692#15.9.13",
    "thirdPartyNotices": "https://go.microsoft.com/fwlink/?LinkId=660708",
    "updateDate": "2019-06-14T10:04:31.4349109Z",
    "catalog": {
      "buildBranch": "d15.9",
      "buildVersion": "15.9.28307.718",
      "id": "VisualStudio/15.9.13+28307.718",
      "localBuild": "build-lab",
      "manifestName": "VisualStudio",
      "manifestType": "installer",
      "productDisplayVersion": "15.9.13",
      "productLine": "Dev15",
      "productLineVersion": "2017",
      "productMilestone": "RTW",
      "productMilestoneIsPreRelease": "False",
      "productName": "Visual Studio",
      "productPatchVersion": "13",
      "productPreReleaseMilestoneSuffix": "1.0",
      "productRelease": "RTW",
      "productSemanticVersion": "15.9.13+28307.718",
      "requiredEngineVersion": "1.18.1049.33485"
    },
    "properties": {
      "campaignId": "1774173123.1508394022",
      "channelManifestId": "VisualStudio.15.Release/15.9.13+28307.718",
      "nickname": "",
      "setupEngineFilePath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\Installer\\vs_installershell.exe"
    }
  }
]
```
#### where msbuild
```
C:\>where msbuild 
C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\MSBuild\15.0\Bin\MSBuild.exe
C:\Windows\Microsoft.NET\Framework\v4.0.30319\MSBuild.exe
```

--- a/src/NoTargets/Sdk/Sdk.targets
+++ b/src/NoTargets/Sdk/Sdk.targets
@@ -72,9 +72,9 @@
   <Import Project="$(CustomAfterNoTargets)" Condition="'$(CustomAfterNoTargets)' != '' and Exists('$(CustomAfterNoTargets)')" />
 
   <!-- 
-    Microsoft.Managed.Targets is imported by the managed language target files, but most of the msbuild tasks are actually in Microsoft.Common.Currentversion.targets.
+    Microsoft.Managed.Targets is imported by the managed language target files in MSBuild 16.0 and above, but most of the msbuild tasks are actually in Microsoft.Common.Currentversion.targets.
     So import it when the managed targets do not get imported.
   -->
-  <Import Project="$(MSBuildToolsPath)\Microsoft.Managed.targets" Condition="'$(ManagedLanguageTargetsGotImported)' != 'true'"/>
+  <Import Project="$(MSBuildToolsPath)\Microsoft.Managed.targets" Condition="'$(MSBuildAssemblyVersion)' >= '16.0' And '$(ManagedLanguageTargetsGotImported)' != 'true'" />
 
 </Project>

